src/aosm/azext_aosm/deploy/pre_deploy.py

```diff
@@ -116,7 +123,7 @@ def ensure_publisher_exists(
                 publisher_name=publisher_name,
                 parameters=Publisher(location=location, scope="Private"),
             )
-            pub.result()
+            LongRunningOperation(self.cli_ctx, "Creating publisher...")(pub)
```

Comment:
Any reason not to call this `poller`?

GitHub is really annoying: you can only add a comment on changed lines +/- 3 lines. So really this comment should be against line 121.